scipy/optimize/_minimize.py

```diff
@@ -696,7 +697,9 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
                 bounds = _remove_from_bounds(bounds, i_fixed)
                 fun = _remove_from_func(fun, i_fixed, x_fixed)
                 if callable(callback):
-                    callback = _remove_from_func(callback, i_fixed, x_fixed)
+                    sig = inspect.signature(callback)
+                    if set(sig.parameters) != {'intermediate_result'}:
+                        callback = _remove_from_func(callback, i_fixed, x_fixed)
```

Comment:
I read your analysis and the idea behind the fix seems correct. So other than the reviews for test I gave above, this PR seems good.